megamek/src/megamek/client/ui/swing/StatusBarPhaseDisplay.java

```diff
@@ -52,14 +54,49 @@
  *
  */
 public abstract class StatusBarPhaseDisplay extends AbstractPhaseDisplay
-        implements ActionListener, KeyListener, IPreferenceChangeListener {
+        implements ActionListener, MouseListener, KeyListener, IPreferenceChangeListener {
 
     private static final long serialVersionUID = 639696875125581395L;
     
     protected static final int TRANSPARENT = 0xFFFF00FF;
     
     protected static final Dimension minButtonSize = new Dimension(32,32);
-    
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+
+    }
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+
+    }
+
+    @Override
+    public void mousePressed(MouseEvent e) {
+
+    }
+
+    @Override
+    public void mouseReleased(MouseEvent e) {
+
+    }
+
+    @Override
+    public void mouseEntered(MouseEvent e) {
+
+    }
+
+    @Override
+    public void mouseExited(MouseEvent e) {
+
+    }
+
+    @Override
+    public void removeAllListeners() {
```

Comment:
Is this particular one necessary?

Reply:
It's part of the MouseListener interface and gives me compile errors if it's not implemented. Is there a better alternate implementation in Java?

Reply:
Just that you don't override it in the other class and it seems like it might have some other unintended side effects. Maybe call super.removeAllListeners() inside?

Reply:
Huh. I might have things mixed up then. I'll try removing it and the overrides again and give it another go.